apps/sms/views/shared/js/activity_handler.js

```diff
@@ -114,43 +117,51 @@ var ActivityHandler = {
           return size;
         }, 0);
 
-        if (size > Settings.mmsSizeLimitation) {
-          Utils.alert({
-            id: 'attached-files-too-large',
-            args: {
-              n: activityData.blobs.length,
-              mmsSize: (Settings.mmsSizeLimitation / 1024).toFixed(0)
-            }
-          }).then(() => ActivityClient.postResult());
-          return;
-        }
+        dataPromise = MozSettingsClient.mmsSizeLimitation().then((limit) => {
```

Comment:
I returned the postError directly in the end because I don't want to wrap the logic into another promise for the error cases, but please let me know if you think it's not appropriate.